### PR TITLE
Mq grass logic

### PR DESCRIPTION
**Pull request by Joshua8600**

### CODE REVIEW

File: packages/core/data/oot/world_mq/deku_tree_mq.yml [hunk 1]
```diff
@@ -18,27 +18,67 @@
...
+    "MQ Deku Tree Grass Room Before Compass 6": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
+    "MQ Deku Tree Grass Room Before Compass 7": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
+    "MQ Deku Tree Grass Basement Lower 1": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 2": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 3": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 4": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Spike 1": "can_cut_grass && (can_use_slingshot || can_use_bow) && (can_use_sticks || has_fire)"
+    "MQ Deku Tree Grass Room Before Spike 2": "can_cut_grass && (can_use_slingshot || can_use_bow) && (can_use_sticks || has_fire)"
+    "MQ Deku Tree Grass Room Before Spike 3": "can_cut_grass && (can_use_slingshot || can_use_bow) && (can_use_sticks || has_fire)"
+    "MQ Deku Tree Grass Room Before Spike 4": "can_cut_grass && (can_use_slingshot || can_use_bow) && (can_use_sticks || has_fire)"
```
Comment (XenoWars) [inline packages/core/data/oot/world_mq/deku_tree_mq.yml:46]: iirc, this is the first room in deku basement. if you can use bow, you dont need sticks or fire cause of the torch. i imagine this was copy/pasted from water room access which would then need sticks or fire for the torches within.
```suggestion
    "MQ Deku Tree Grass Room Before Spike 4": "can_cut_grass && (can_use_bow || (can_use_slingshot && (can_use_sticks || has_fire))"
```

File: packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml [hunk 2]
```diff
@@ -23,3 +23,7 @@
     "MQ Bottom of the Well Pot Side Room 2": "can_play_zelda || (has_explosives_or_hammer && has_hover_boots)"
     "MQ Bottom of the Well Pot Side Room 3": "can_play_zelda || (has_explosives_or_hammer && has_hover_boots)"
     "MQ Bottom of the Well Pot Lobby Alcove": "has_explosives_or_hammer && can_hit_triggers_distance"
+    "MQ Bottom of the Well Grass Dead-Hand 1": "can_cut_grass && (has_ranged_weapon_child || has_ranged_weapon_adult || has_explosives_or_hammer || can_use_sticks || can_use_sword)" #Many ways to lower the water.
```
Comment (XenoWars) [inline packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml:26]: Age swap logic hasn't been done for MQ yet, but you can compact the ranged weapons into a single macro. "has_weapon_ranged" i believe it is.

Also, can child reach this with broken giant's knife?

File: packages/core/data/oot/world_mq/deku_tree_mq.yml [hunk 3]
```diff
@@ -56,8 +104,25 @@
     DEKU_BLOCK: "true"
   locations:
     "MQ Deku Tree Scrub": "can_hit_scrub && scrub_price(0x1b)"
+    "MQ Deku Tree Grass Basement Upper 1": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Upper 2": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Upper 3": "can_cut_grass"
+    "MQ Deku Tree Grass Larvae Room 1": "is_child && can_cut_grass" #You can go backwards to get this specific grass with nothing else.
+    "MQ Deku Tree Grass Larvae Room 2": "is_child && can_cut_grass" #You can go backwards to get this specific grass with nothing else.
```
Comment (XenoWars) [inline packages/core/data/oot/world_mq/deku_tree_mq.yml:111]: put these in backrooms, not here. could go through the tunnel then age swap to adult.

File: packages/core/data/oot/world_mq/deku_tree_mq.yml [hunk 4]
```diff
@@ -56,8 +104,25 @@
...
+    "MQ Deku Tree Grass Gravestone Room 1": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 2": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 3": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 4": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 5": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Back Room 1": "is_child && can_cut_grass && (can_use_sticks || has_fire)"
+    "MQ Deku Tree Grass Back Room 2": "is_child && can_cut_grass && (can_use_sticks || has_fire)"
+    "MQ Deku Tree Grass Back Room 3": "is_child && can_cut_grass && (can_use_sticks || has_fire)"
```
Comment (XenoWars) [inline packages/core/data/oot/world_mq/deku_tree_mq.yml:119]: likewise, these ones dont need to be here.

File: packages/core/data/oot/world_mq/jabu_jabu_mq.yml [hunk 5]
```diff
@@ -34,6 +36,15 @@
     "MQ Jabu-Jabu Pot Underwater Alcove 2": "is_child || (is_adult && can_dive_small)" #Future-proofing this for eventual ageless bow and slingshot. Adult cannot dive to the alcove without a scale.
     "MQ Jabu-Jabu Pot Boomerang Room 1": "is_child || (is_adult && can_dive_small)" #Future-proofing this for eventual ageless bow and slingshot. Adult cannot dive to the alcove without a scale.
     "MQ Jabu-Jabu Pot Boomerang Room 2": "is_child || (is_adult && can_dive_small)" #Future-proofing this for eventual ageless bow and slingshot. Adult cannot dive to the alcove without a scale.
+    "MQ Jabu-Jabu Grass Main Room Top 1": "(is_child || (is_adult && can_dive_small)) && can_cut_grass && has_explosives_or_hammer"
+    "MQ Jabu-Jabu Grass Main Room Top 2": "(is_child || (is_adult && can_dive_small)) && can_cut_grass && has_explosives_or_hammer"
```
Comment (XenoWars) [inline packages/core/data/oot/world_mq/jabu_jabu_mq.yml:40]: why do these need boulder destroying ability? or diving?

**Review by XenoWars — CHANGES_REQUESTED** [review]

File: packages/core/data/oot/world_mq/deku_tree_mq.yml (hunk 3 above)
Comment (Joshua8600) [reply packages/core/data/oot/world_mq/deku_tree_mq.yml]: Fixed.

File: packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml (hunk 2 above)
Comment (Joshua8600) [reply packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml]: Fixed the macros issue.

I am unsure about the broken giant's knife though. It is basically the Kokiri Sword so I would assume so?

Comment (XenoWars) [reply packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml]: BGK is shorter than the dagger

File: packages/core/data/oot/world_mq/jabu_jabu_mq.yml (hunk 5 above)
Comment (XenoWars) [reply packages/core/data/oot/world_mq/jabu_jabu_mq.yml:40]: ah, so they're in the room with all the holes

File: packages/core/data/oot/world_mq/deku_tree_mq.yml [hunk 6]
```diff
@@ -56,8 +106,20 @@
...
   locations:
     "MQ Deku Tree Scrub": "can_hit_scrub && scrub_price(0x1b)"
+    "MQ Deku Tree Grass Basement Upper 1": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Upper 2": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Upper 3": "can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 1": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 2": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 3": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 4": "is_child && can_cut_grass"
+    "MQ Deku Tree Grass Gravestone Room 5": "is_child && can_cut_grass"
```
Comment (XenoWars) [inline packages/core/data/oot/world_mq/deku_tree_mq.yml:116]: Meant to remove these gravestone ones from here too since they are in backrooms.

**Review by XenoWars — CHANGES_REQUESTED** [review]

File: packages/core/data/oot/world_mq/deku_tree_mq.yml (hunk 6 above)
Comment (Joshua8600) [reply packages/core/data/oot/world_mq/deku_tree_mq.yml]: done.

File: packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml (hunk 2 above)
Comment (Joshua8600) [reply packages/core/data/oot/world_mq/bottom_of_the_well_mq.yml]: BGK cannot reach it.

**Pull request merged**